src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -57,19 +111,87 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
+        public async IAsyncEnumerable<ConfigurationClient> GetAvailableClients([EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
-        }
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+            DateTimeOffset minimalRefreshTime = _lastFallbackClientRefreshAttempt + MinimalClientRefreshInterval;
+
+            // Principle of refreshing fallback clients:
+            // 1.Perform initial refresh attempt to query available clients on app start-up.
+            // 2.Refreshing when cached fallback clients have expired, FallbackClientRefreshExpireInterval is due.
+            // 3.At least wait MinimalClientRefreshInterval between two attempt to ensure not perform refreshing too often.
+
+            if (_replicaDiscoveryEnabled &&
+                now >= minimalRefreshTime && 
+                (_dynamicClients == null ||
+                now >= _lastFallbackClientRefresh + FallbackClientRefreshExpireInterval))
+            {
+                _lastFallbackClientRefreshAttempt = now;
+
+                await RefreshFallbackClients(cancellationToken).ConfigureAwait(false);

```

Comment:
Even though we keep it around, I feel we should max this call out given it's not a necessity to continue on. Perhaps using `MinimalClientRefreshInterval` as the max time allowance?